src/test/ui/lifetimes/lifetime-errors/ex1-return-one-existing-name-if-else-using-impl.nll.stderr

```diff
@@ -7,7 +7,7 @@ LL |     fn foo<'a>(x: &i32, y: &'a i32) -> &'a i32 {
    |            lifetime `'a` defined here
 LL | 
 LL |         if x > y { x } else { y }
-   |                    ^ function was supposed to return data with lifetime `'a` but it is returning data with lifetime `'1`
+   |                    ^ method was supposed to return data with lifetime `'a` but it is returning data with lifetime `'1`
```

Comment:
Hmm... actually, this is not a method... all the others appear correct.

And I also just realized, that I don't know why this change in UI happened? Any ideas?

Reply:
The change causing this is here: https://github.com/rust-lang/rust/pull/67742/files#diff-5f616dd3631e07b51dd7c59f32ccfb76R433

Reply:
Hmm... do you mean this one instead?

https://github.com/rust-lang/rust/pull/67742/files?file-filters%5B%5D=.rs#diff-5f616dd3631e07b51dd7c59f32ccfb76R505-R507

Also, this seems to imply that we are getting a `DefKind::Method` instead of a `DefKind::Fn`. Is this expected?

Reply:
`DefKind::Method` is for all associated functions. This is generally what `Method` means in the compiler, not that it should be leaking into error messages.

Reply:
We should rename `DefKind::Method` to `DefKind::AssocFn` so that it fits with the scheme used for `AssocTy`, `AssocConst`, etc. cc @petrochenkov @eddyb

Reply:
Pretty sure we have an issue open about this, yeah.